src/oop/special_class_method/hard/q10/MainTest.java

```diff
@@ -0,0 +1,192 @@
+                "Your " + className + " class is missing the " + firstFieldName + " attributes.");
+        assertTrue(testClass.hasModifier(firstFieldName, "private"),
+                "Your " + firstFieldName + " does not have the correct visibility modifier.");
+        assertTrue(testClass.hasField(secondFieldName, String.class),
+                "Your " + className + " class is missing the " + secondFieldName + " attributes.");
+        assertTrue(testClass.hasModifier(secondFieldName, "private"),
+                "Your " + secondFieldName + " does not have the correct visibility modifier.");
+    }
+
+    private static Stream<Arguments> inputProvider() {
+        return Stream.of(
+                Arguments.of("First", "Second"),
+                Arguments.of("Science", "Anna"),
+                Arguments.of("Jack", "History"),
+                Arguments.of("Finn", "Fiction")
+
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    public void awardClassHasCorrectConstructor(String value1, String value2) throws Throwable {
+        Object[][] arguments = {
+                {value1, String.class},
+                {value2, String.class}
+        };
+        Class<?>[] constructorClasses = {String.class, String.class};
+        Object classInstance = testClass.createInstance(arguments);
+        String incorrectDoubleValueMessage =
+                "Your " + className + " constructor does not initialize the " + firstFieldName + " attribute to the correct value.";
+        String incorrectStringValueMessage =
+                "Your " + className + " constructor does not initialize the " + secondFieldName + " attribute to the correct value.";
+        String incorrectConstructorVisibilityModifier =
+                "Your " + className + " constructor does not have the correct visibility modifier.";
+        _assertEquals(value1, testClass.getFieldValue(classInstance, firstFieldName),
+                incorrectDoubleValueMessage);
+        _assertEquals(value2, testClass.getFieldValue(classInstance, secondFieldName),
+                incorrectStringValueMessage);
+        assertTrue(testClass.hasModifier(constructorClasses, "public"), incorrectConstructorVisibilityModifier);
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    public void awardClassHasCorrectSetAwardRecipientMethod(String initialValue, String setValue) throws Throwable {
+        Object[][] arguments = {
+                {initialValue, String.class},
+                {"", String.class}
+        };
+        Object classInstance = testClass.createInstance(arguments);
+        Object[][] setSizeArguments = {
+                {setValue, String.class}
+        };
+        String[] setMethodModifiers = {"public"};
+        String incorrectSetMethodMessage = String.join(" ",
+                "Your", setFirstMethodName, "does not correctly set the value of the", firstFieldName, "attribute.");
```

Comment:
make sure all the error messages where you include a method name has "method" after it